src/hooks/useCampaign.js

```diff
@@ -7,8 +7,8 @@ export const useCampaign = () => {
   const [{ provider }, dispatch] = useSharedState();
 
   const getAdvertiserPayouts = async (campaignAddress) => {
-    const signer = await provider.getSigner();
-    const Campaign = new Contract(campaignAddress, CampaignJson, signer);
+    const signer = await provider?.getSigner();
```

Comment:
Can you define `signer` and `Campaign` once in this file?

Reply:
i cant use await outside of that function